### PR TITLE
Started to implement inventory handling

### PR DESCRIPTION
**Pull request by xPenguinx**
I have been working on Inventories started by handling the OpenWindow and WindowItems packets to save inventory data. I just wanted to commit this so others can help along the way its nowhere near done but progress. The plan is to have an array of opened inventories and when edited, edit the items in it via the id using Dictionary<Item,Slot#> 

The WindowItems packet is not  fully finished as the ReadNextNBT is causing some issues but wanted to share my contributions so others can help further / fix you can ofc comment it for now until a further fix is found

### CODE REVIEW

File: MinecraftClient/Protocol/Handlers/Protocol18.cs [hunk 1]
```diff
@@ -464,6 +464,41 @@ internal bool HandlePacket(int packetID, List<byte> packetData)
...
+                            {
+                                short itemID = dataTypes.ReadNextShort(packetData);
+                                if (itemID == -1) continue;
+                                byte itemCount = dataTypes.ReadNextByte(packetData);
+                                short itemDamage = dataTypes.ReadNextShort(packetData);
+                                Item item = new Item(itemID, itemCount, itemDamage, 0);
+                                //TODO: Add to the dictionary for the inventory its in using the id
+                                if (packetData.ToArray().Count() > 0)
+                                {
+                                    dataTypes.ReadNextNbt(packetData);
```
Comment (xPenguinx) [inline MinecraftClient/Protocol/Handlers/Protocol18.cs]: Causing a crash for some nbt tags comment it till a fix for ReadNextNBT is found or something

**Comment by ORelio** [comment]
Looks good! Before merging, would it be possible to add a setting similar to `terrainandmovements` such as `inventoryhandling` (or other explicit name) with default value of `false`? This would help in implementing and testing the feature without breaking stability of development builds for other users 🙂

**Comment by xPenguinx** [comment]
> Looks good! Before merging, would it be possible to add a setting similar to `terrainandmovements` such as `inventoryhandling` (or other explicit name) with default value of `false`? This would help in implementing and testing the feature without breaking stability of development builds for other users 🙂

good idea

**Comment by ORelio** [comment]
All good! Merged. 😉

**Pull request merged**